components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -592,15 +591,14 @@ SUBROUTINE p3_main_main_loop(kts, kte, kbot, ktop, kdir, log_predictNc, dt, odt,
       if (qitot(k).ge.qsmall) log_exitlevel = .false.
       !enddo
       if (log_exitlevel .and.                                                           &
-           ((t(k).lt.zerodegc .and. supi(k).lt.-0.05_rtype) .or.                              &
-           (t(k).ge.zerodegc .and. sup(k) .lt.-0.05_rtype))) goto 555   !i.e. skip all process rates
+         (t(k).lt.zerodegc .and. supi(k).lt.-0.05_rtype)) goto 555   !i.e. skip all process rates
```

Comment:
I think we want an "or t(k).ge.zerodegc" clause here - I think you're skipping all warm-rain microphysics all the time, right?

Reply:
No, this line means there is negligible mass/number conc., and if we have dry atmosphere as well - we skip even nucleation (liquid / ice). 
But we do not have explicit nucleation in positive temp (this is the point of this PR), and we do not have an Sw field, so it means it will keep calculating the warm-phase processes only when there is sufficient mass/number conc. no matter if it's a dry OR wet atmosphere.
Make sense?

Reply:
Oh, I somehow missed the log_exitlevel part of the if statement. Ok, I agree that this is right. Thanks for explaining.